cmake/CheckCopyrightScript.cmake

```diff
@@ -0,0 +1,68 @@
+#=========================================================================================
+# (C) (or copyright) 2020-2021. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# This script checks that a file has an up to date copyright. Only runs for
+# users with @lanl.gov email addresses.
+
+set(SOURCE_DIR ${CMAKE_ARGV3})
+set(GIT_EXECUTABLE ${CMAKE_ARGV4})
+set(INPUT ${CMAKE_ARGV5})
+
+string(TIMESTAMP CURRENT_YEAR "%Y")
+
+# Check if the file has an up to date copyright
+file(READ ${SOURCE_DIR}/${INPUT} CONTENTS)
+string(REGEX MATCH "\\(C\\) \\(or copyright\\) 2020-${CURRENT_YEAR}\\. Triad National Security, LLC\\. All rights reserved\\." HAS_COPYRIGHT ${CONTENTS})
```

Comment:
So if I add a new file this year to Parthenon is the copyright still supposed to start from 2020 for that file?

Reply:
Ah, good point...